subcommand/cert-manager/command.go

```diff
@@ -0,0 +1,243 @@
+package certmanager
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/consul-k8s/helper/cert"
+	"github.com/hashicorp/consul-k8s/subcommand/flags"
+	"github.com/mitchellh/cli"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Command struct {
+	flagSet   *flag.FlagSet
+	httpFlags *flags.HTTPFlags
+	UI        cli.Ui
+
+	flagWebhookName     string // MutatingWebhookConfiguration for updating
+	flagAutoHosts       string // SANs for the auto-generated TLS cert.
+	flagSecretName      string // Name of secret where certificates will be written to.
+	flagSecretNamespace string // Namespace of the secret where certificates will be written to.
+	flagCertFile        string // TLS cert for listening (PEM)
+	flagKeyFile         string // TLS cert private key (PEM)
+
+	clientset kubernetes.Interface
+
+	once sync.Once
+	help string
+}
+
+func (c *Command) init() {
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.StringVar(&c.flagWebhookName, "webhook-name", "",
+		"MutatingWebhookConfiguration name. If specified, will auto generate cert bundle.")
+	c.flagSet.StringVar(&c.flagAutoHosts, "tls-hosts", "",
+		"Comma-separated hosts for auto-generated TLS cert. If specified, will auto generate cert bundle.")
+	c.flagSet.StringVar(&c.flagSecretName, "secret-name", "",
+		"Name of the secret to update TLS certificates")
+	c.flagSet.StringVar(&c.flagSecretNamespace, "secret-namespace", "default",
+		"Namespace of the secret to update TLS certificates. Defaults to default")
+	c.flagSet.StringVar(&c.flagCertFile, "tls-cert-file", "",
+		"PEM-encoded TLS certificate to serve. If blank, will generate random cert.")
+	c.flagSet.StringVar(&c.flagKeyFile, "tls-key-file", "",
+		"PEM-encoded TLS private key to serve. If blank, will generate random cert.")
+
+	c.httpFlags = &flags.HTTPFlags{}
+	flags.Merge(c.flagSet, c.httpFlags.Flags())
+	c.help = flags.Usage(help, c.flagSet)
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+	if err := c.flagSet.Parse(args); err != nil {
+		c.UI.Error(fmt.Sprintf("parsing flagSet: %s", err))
+		return 1
+	}
+	if len(c.flagSet.Args()) > 0 {
+		c.UI.Error(fmt.Sprintf("invalid arguments: %s", errors.New("should have no non-flag arguments")))
+		return 1
+	}
+
+	if c.flagSecretName == "" {
+		c.UI.Error(fmt.Sprintf("invalid arguments: %s", errors.New("secret-name must be set")))
```

Comment:
```suggestion
		c.UI.Error(fmt.Sprintf("invalid arguments: %s", errors.New("-secret-name must be set")))
```

Curious why you want the `fmt` and `errors.New` instead of
```
		c.UI.Error("invalid arguments: -secret-name must be set")
```

Reply:
Done whilst in a hurry. Cleaning up :)